contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,396 @@
+        generalParameters.withdrawRequestEnablePeriod = _withdrawRequestEnablePeriod;
+        emit SetWithdrawRequestParams(_withdrawRequestPendingPeriod, _withdrawRequestEnablePeriod);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setClaimFee}. */
+    function setClaimFee(uint256 _fee) external onlyOwner {
+        generalParameters.claimFee = _fee;
+        emit SetClaimFee(_fee);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setWithdrawSafetyPeriod}. */
+    function setWithdrawSafetyPeriod(uint256 _withdrawPeriod, uint256 _safetyPeriod) external onlyOwner { 
+        if (_withdrawPeriod < 1 hours) revert WithdrawPeriodTooShort();
+        if (_safetyPeriod > 6 hours) revert SafetyPeriodTooLong();
+        generalParameters.withdrawPeriod = _withdrawPeriod;
+        generalParameters.safetyPeriod = _safetyPeriod;
+        emit SetWithdrawSafetyPeriod(_withdrawPeriod, _safetyPeriod);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setHatVestingParams}. */
+    function setHatVestingParams(uint256 _duration, uint256 _periods) external onlyOwner {
+        if (_duration >= 180 days) revert HatVestingDurationTooLong();
+        if (_periods == 0) revert HatVestingPeriodsCannotBeZero();
+        if (_duration < _periods) revert HatVestingDurationSmallerThanPeriods();
+        generalParameters.hatVestingDuration = _duration;
+        generalParameters.hatVestingPeriods = _periods;
+        emit SetHatVestingParams(_duration, _periods);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setMaxBountyDelay}. */
+    function setMaxBountyDelay(uint256 _delay) external onlyOwner {
+        if (_delay < 2 days) revert DelayTooShort();
+        generalParameters.setMaxBountyDelay = _delay;
+        emit SetMaxBountyDelay(_delay);
+    }
+
+    /** @notice See {IHATVaultsRegistry-createVault}. */
+    function createVault(
+        IERC20 _asset,
+        address _owner,
+        address _committee,
+        IRewardController _rewardController,
+        uint256 _maxBounty,
+        HATVault.BountySplit memory _bountySplit,
+        string memory _descriptionHash,
+        uint256[2] memory _bountyVestingParams,
+        bool _isPaused
+    ) 
+    external 
+    returns(address vault)
+    {
+        vault = Clones.clone(hatVaultImplementation);
+
+        HATVault(vault).initialize(
+            IHATVault.VaultInitParams({
+                rewardController: _rewardController,
+                vestingDuration: _bountyVestingParams[0],
+                vestingPeriods: _bountyVestingParams[1],
+                maxBounty: _maxBounty,
+                bountySplit: _bountySplit,
+                asset: _asset,
+                owner: _owner,
+                committee: _committee,
+                isPaused: _isPaused
+            })
+        );
+
+        hatVaults.push(vault);
+
+        emit VaultCreated(
+            vault,
+            address(_asset),
+            _committee,
+            _rewardController,
+            _descriptionHash,
```

Comment:
it did bother me that it's not the same order as the func params but that's just ocd i guess :)

Reply:
very minor but valid point, changing this in https://github.com/hats-finance/hats-contracts/pull/340